packages/hash/frontend/src/pages/[account-slug]/types/entity-type/[entity-type-id].page.tsx

```diff
@@ -85,10 +94,23 @@ const Page: NextPageWithLayout = () => {
       data.properties.map((property) => {
         const propertyKey = extractBaseUri(property.$id);
 
-        return [propertyKey, { $ref: property.$id }];
+        const prop: ValueOrArray<PropertyTypeReference> = property.array
+          ? {
+              type: "array",
+              minItems: property.minValue,
+              maxItems: property.maxValue,
+              items: { $ref: property.$id },
+            }
+          : { $ref: property.$id };
+
+        return [propertyKey, prop];
       }),
     );
 
+    const required = data.properties
+      .filter((property) => property.required)
+      .map((property) => `${property.$id.split("/v/")[0]}/`); // @todo: is there a better way to do this?
```

Comment:
Is this `extractBaseUri`?

Reply:
Yes, I missed it, Thanks. 98b5bd8e0a997e22cfbc6e068c6d1ebbe2b360c6